Mage/src/main/java/mage/abilities/common/EntersBattlefieldOneOrMoreTriggeredAbility.java

```diff
@@ -40,37 +40,32 @@ private EntersBattlefieldOneOrMoreTriggeredAbility(final EntersBattlefieldOneOrM
 
     @Override
     public boolean checkEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.ZONE_CHANGE_GROUP;
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE_BATCH;
     }
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        ZoneChangeGroupEvent zEvent = (ZoneChangeGroupEvent) event;
+
         Player controller = game.getPlayer(this.controllerId);
-        if (zEvent.getToZone() != Zone.BATTLEFIELD || controller == null) {
+        if (controller == null) {
             return false;
         }
 
+        ZoneChangeBatchEvent zEvent = (ZoneChangeBatchEvent) event;
+        Stream<Permanent> enteringPermanents = zEvent.getEvents().stream()
```

Comment:
Do not use stream fields/params due code style. Use filter with all conditions inside.